src/dataset-builder/ConceptSearch.ts

```diff
@@ -0,0 +1,130 @@
+  const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
+    props;
+  const [search, setSearch] = useState<string>(initialSearch);
+  const [cart, setCart] = useState<Concept[]>(initialCart);
+  const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
+  useEffect(() => {
+    void searchConcepts(() => {
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+    });
+  }, [search, datasetId, domainOption.root, searchConcepts]);
+  return h(Fragment, [
+    div({ style: { padding: `${PAGE_PADDING_HEIGHT}rem ${PAGE_PADDING_WIDTH}rem` } }, [
+      h2({ style: { display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            onClick: onCancel,
+            'aria-label': 'cancel',
+          },
+          [icon('left-circle-filled', { size: 32 })]
+        ),
+        div({ style: { marginLeft: 15 } }, [domainOption.category]),
+      ]),
+      h(StringInput, {
+        onChange: (value: string) => {
+          setSearch(value);
+        },
+        value: search,
+        placeholder: 'Search',
+      }),
+      concepts.status === 'Ready'
+        ? h(SimpleTable, {
+            'aria-label': 'concept search results',
+            underRowKey: 'underRow',
+            rowStyle: { marginTop: 5, marginBottom: 5 },
+            headerRowStyle: {
+              height: '100%',
+              display: 'flex',
+              paddingTop: 15,
+              paddingBottom: 15,
+              backgroundColor: colors.light(0.4),
+              borderRadius: '8px 8px 0px 0px',
+              border: `.5px solid ${colors.dark(0.2)}`,
+            },
+            columns: [
+              { header: 'Concept Name', width: 710, key: 'name' },
+              { header: 'Concept ID', width: 195, key: 'id' },
+              { header: 'Code', width: 195, key: 'code' },
+              { header: 'Roll-up count', width: 205, key: 'count' },
+              { width: 100, key: 'hierarchy' },
+            ],
+            rows: _.map((concept) => {
+              const [label, iconName]: [string, IconId] = (() => {
+                if (_.contains(concept, cart)) {
+                  return ['remove', 'minus-circle-red'];
+                }
+                return ['add', 'plus-circle-filled'];
+              })();
+              return {
+                name: div({ style: { display: 'flex' } }, [
+                  h(Link, { 'aria-label': label, onClick: () => setCart(_.xor(cart, [concept])) }, [
+                    icon(iconName, { size: 16 }),
+                  ]),
+                  div({ style: { marginLeft: 5 } }, [concept.name]),
+                ]),
+                id: concept.id,
+                count: concept.count,
+                hierarchy: div({ style: { display: 'flex' } }, [
+                  h(
+                    Link,
+                    {
+                      'aria-label': label,
```

Comment:
Should this have the same label? What do you think about `open hierarchy for ${label}`

Reply:
You're right, I've changed the labels here to include the concept ID.